streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -286,7 +305,7 @@ private void restoreBatch(final Collection<ConsumerRecord<byte[], byte[]>> batch
 
     @Override
     public void evictWhile(final Supplier<Boolean> predicate,
-                           final Consumer<KeyValue<Bytes, ContextualRecord>> callback) {
+                           final Consumer<Eviction<K, V>> callback) {
```

Comment:
I added the `Eviction` class so we can pack up all three pieces (key, value, context) to give to the callback. We can't use ContextualRecord anymore, since it's a post-serialization container for the value.